support/README.md

```diff
@@ -0,0 +1,4 @@
+# Support tools
+
+A collection of helper scripts meant to be used by the Envoy developers. For
+example, scripts executed as post-commit hooks would go here.
```

Comment:
Would it be worth it to add a little more text here of which things are installed and roughly what they do? I might mention that when the check format script is installed the developer can use the `-n` flag with `git commit` to avoid checking if they don't want to temporarily. Some people might get annoyed if they don't know this.

Also, perhaps we should link to here from the top level developer guide? https://github.com/envoyproxy/envoy/blob/master/DEVELOPER.md or even https://github.com/envoyproxy/envoy#contributing?

Reply:
@mattklein123 Sounds good, let's do all of those things. I'll take another pass later today.